lib/31XUtils/schemaUtils31X.js

```diff
@@ -50,6 +50,35 @@ module.exports = {
     return isTypeMatching;
   },
 
+  /**
+   * Identifies the correct type from the list according to the example
+   *
+   * @param {*} example a provided example value
+   * @param {Array} typesArray the types array of the schema
+   * @returns {string} the most according type to the example
+   */
+  findTypeByExample(example, typesArray) {
+    const mapTypes = {
+      'boolean': 'boolean',
+      'null': 'null',
+      'undefined': 'null',
+      'number': 'number',
+      'bigInt': 'number',
+      'object': 'object',
+      'string': 'string'
+    };
+    let exampleType = typeof example,
+      mappedType = mapTypes[exampleType];
+
+    if (mappedType === 'number') {
+      mappedType = Number.isInteger(example) ? 'integer' : mappedType;
+    }
+
+    return typesArray.find((type) => {
+      return type === mappedType;
+    });
```

Comment:
should we consider this scenario?
```
type: 
  - string
  - integer
examples:
  - true
  - 'test'
```
  There is a valid example but it won't be considered with this implementation

Reply:
Added scenario in the latest commit